.gitlab-ci.yml

```diff
@@ -0,0 +1,169 @@
+# I don't know what the minimum size is, but we cannot build on t3.micro.
+
+# TODO: Factor common builds between different tests.
+
+# The parameters for our job matrix:
+#
+# 1. Generator (Make, Ninja, MSBuild)
+# 2. Compiler (GCC, Clang, MSVC)
```

Comment:
what versions of the compilers? Can we have builds against a few different versions?

Reply:
The versions are determined by the image. It is definitely possible to set up jobs that use different images to test against different versions of compilers.

The only editable description of an image on Docker Hub is its tag, but it is not possible to encode, in a human-readable way, all the metadata about an image, including versions of dependencies installed, in the tag. We will have to maintain a separate catalog of images until Docker Hub expands its capabilities. If we build images out of [rippled-docker](https://docs.gitlab.com/ee/ci/ci_cd_for_external_repos/github_integration.html), we can put that catalog in the README.